src/network/networkd-link.c

```diff
@@ -3288,15 +3288,19 @@ int link_update(Link *link, sd_netlink_message *m) {
         if (carrier_gained) {
                 log_link_info(link, "Gained carrier");
 
-                r = link_carrier_gained(link);
-                if (r < 0)
-                        return r;
+                if (link->state != LINK_STATE_CONFIGURED || !link->network->configure_without_carrier) {
+                        r = link_carrier_gained(link);
+                        if (r < 0)
+                                return r;
+                }
         } else if (carrier_lost) {
                 log_link_info(link, "Lost carrier");
 
-                r = link_carrier_lost(link);
-                if (r < 0)
-                        return r;
+                if (link->state != LINK_STATE_CONFIGURED || !link->network->configure_without_carrier) {
+                        r = link_carrier_lost(link);
+                        if (r < 0)
+                              return r;
+                }
```

Comment:
i think it would make a lot of sense to move this conditions into the top of link_carrier_lost() instead. i.e. call the function here whenever the carrier changes, but then exit early from it when we actually don't care about carrier changes...

Reply:
I made the changes as asked by @poettering

But I am not sure if this change affects link_carrier_reset() which first calls link_carrier_lost() and then link_carrier_gained(). i.e. with new patch link_carrier_reset() will not actually happen. But then may be its not required to be either?